zerver/tornado/event_queue.py

```diff
@@ -217,6 +221,11 @@ def accepts_event(self, event: Mapping[str, Any]) -> bool:
             # delivered if the stream_typing_notifications
             # client_capability is enabled, for backwards compatibility.
             return self.stream_typing_notifications
+        if event["type"] in ["update_display_settings", "update_global_notifications"]:
+            # 'update_display_settings' and 'update_global_notifications'
+            # events are sent only if user_settings_object is False,
+            # otherwise only 'user_settings' are sent.
+            return not self.user_settings_object
```

Comment:
I think where we landed with how this should work is that we can put this at the top as:
```
if not self.user_settings_object and event["type"] in ["update_display_settings", "update_global_notifications"]:
    return False
```

Reply:
@timabbott  As per the chat in chat.zulip.org, 
> We add a user_settings_object entry to client_capabilities, which if sent, results in the update_global_notifications 
> and update_display_settings objects being ignored. (We need this because of how some clients request "all event > types", and we'd like the "all" to be able to not include the deprecated legacy types at all on a modern client).

which means that this should return false (i.e. not accept the event) if `user_settings_object` is true and this makes sense with the variable name `user_settings_object` as well, I think, I will wait for the reply before changing this.

Reply:
Yes, sorry I meant:
```
if self.user_settings_object and event["type"] in ["update_display_settings", "update_global_notifications"]:
    return False
```

which is what you described; the `not` there was a typo.